WordPress/gutenberg
# [RNMobile] Remove unnecessary negative margin around empty gallery block

* `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5386

## What?

This PR removes some unwanted negative margin space from empty gallery blocks on native.

## Why?

A change that was introduced in https://github.com/WordPress/gutenberg/pull/45891, added a `<BlockListCompact>` component to empty gallery blocks via [this line of code](https://github.com/WordPress/gutenberg/blob/e38f53c86e2c1c9cdb581c0b026daf014799c3b2/packages/block-library/src/gallery/edit.js#L499).

That change caused some styling issues on Android as, by default, a negative margin is applied to the added component:

https://github.com/WordPress/gutenberg/blob/fb987952510ef8d55461fd37bcee88273f0ad07a/packages/block-editor/src/components/block-list/block-list-compact.native.js#L44-L45

## How?

The values being passed to the `marginHorizontal` and `marginVertical` props from the gallery block have been set to zero.

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/211904674-cf073df9-e475-449a-91b2-f40ce75797db.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/211904547-c7f5e601-9410-49c8-bf92-40e54180fd1d.png" width="100%"> |

## Testing Instructions

1. Navigate to a new post/page in the Android or iOS app.
2. Add a new gallery block, without uploading any images to it.
3. Observe that the block is styled as expected, with no overlapping border. 
4. Verify that no other regressions have been introduced following this PR by going through some regular flows e.g. uploading a few images and confirming no style regressions, rotating the device to landscape mode, etc.
